core/src/HIP/Kokkos_HIP_IsXnack.hpp

```diff
@@ -43,7 +40,21 @@ be modified if a better way is determined to check for HMM support in Linux.
 Checking for CONFIG_HMM was considered, but it was not present on El Capitan,
 so we infer its presence is not necessary.
 */
-bool xnack_enabled();
+
+// Returns true iff we detect HSA_XNACK=1 in the environment.
+bool xnack_environment_enabled();
+// Returns true iff we detect CONFIG_HMM_MIROR=y in /boot/config-$(uname -r).
+bool xnack_boot_config_has_hmm_mirror();
+// Returns true iff the architecture of the gpu supports accessing system
+// allocated memory
+constexpr bool gpu_arch_can_access_system_allocations() {
+#if defined Kokkos_ARCH_AMD_GFX908 || Kokkos_ARCH_AMD_GFX90A || \
+    Kokkos_ARCH_AMD_GFX942 || Kokkos_ARCH_AMD_GFX942_APU
+  return true;
+#else
+  return false;
+#endif
```

Comment:
This one lives here as it is currently only used for the xnack/hmm printing.

Reply:
Why is this not `Kokkos::SpaceAccesibility<Kokkos::HIP, Kokkos::HostSpace>`? Why do we want to report something we don't support?

Reply:
because this will be true independent of how we set the accessibility in `Kokkos`. If you have a gpu of these archs and you set your system accordingly you will be able to access system allocated memory no matter if we want to mimic that in our `SpaceAccessibility`